internal/controller/promotionstrategy_controller.go

```diff
@@ -372,9 +375,12 @@ func (r *PromotionStrategyReconciler) updatePreviousEnvironmentCommitStatus(ctx
 			commitStatusPhase = promoterv1alpha1.CommitPhaseSuccess
 		}
 
-		err := r.createOrUpdatePreviousEnvironmentCommitStatus(ctx, ctpMap[environment.Branch], commitStatusPhase, previousEnvironmentStatus)
-		if err != nil {
-			return fmt.Errorf("failed to create or update previous environment commit status for branch %s: %w", environment.Branch, err)
+		if previousEnvironmentIndex != -1 && (len(ps.Spec.ActiveCommitStatuses) != 0 || len(ps.Spec.Environments[previousEnvironmentIndex].ActiveCommitStatuses) != 0) {
+			// If there is no configured active checks we should not create a commit status for the previous environment.
```

Comment:
```suggestion
			// Since there is at least one configured active check, and since this is not the first environment,
			// we should not create a commit status for the previous environment.
```